src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,56 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.strategy.InterfaceImplementationFinderStrategy;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = InterfaceImplementationFinderStrategy.getImplementation(interfaceClazz);
```

Comment:
Make sure to check if the class returned by `InterfaceImplementationFinderStrategy.getImplementation(interfaceClazz)` has the `@Component` annotation before proceeding with instance creation.